### ISSUE TITLE
[Uldaman] Altar of Archaedas can be used after Archaedas is dead, locking players inside

### ISSUE THREAD

**Issue by syssneck** [azerothcore/azerothcore-wotlk#7358]
### What client do you play on?

enUS

### Faction

- [ ] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [X] 40-49
- [ ] 50-59

### Current Behaviour

Original report: [Chromie issue#1435](https://github.com/chromiecraft/chromiecraft/issues/1435)

Pedestal in Uldaman can be used after kill boss, as a result of which the doors are **closed forever** and players cannot access the quest room!
![WoWScrnShot_081521_171251](https://user-images.githubusercontent.com/21084600/129481734-18b0c5cd-9633-404d-a940-dafd821d3009.jpg)

### Expected Blizzlike Behaviour

once one time can be used

### Steps to reproduce the problem

use Pedestal 
kill boss
use Pedestal 

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/2d77fe9274005a99f8445b8dac9dd0fcd3faf3c5

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-CustomWorldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-TempAnnouncements](https://github.com/55Honey/Acore_TempAnnouncements)

### Customizations

None

### Server

ChromieCraft

Confirmed on local AC (da1fd68).
After using the Altar of Archaedas after Archaedas' death, both doors close and players can't go back or go to the chest room.
Make it so, that the Altar is unusable, after the boss spawned.

**Comment by Steelcold** [comment]
You can also [kill the boss Ironaya](https://imgur.com/a/tKZdzrg) without opening the door, using your pet (if you got one).

**Comment by syssneck** [comment]
@Steelcold yes, but that is another issue, see here: https://github.com/azerothcore/azerothcore-wotlk/issues/7465
It has been fixed (not yet live on CC).

**Comment by Si1ker** [comment]
> You can also [kill the boss Ironaya](https://imgur.com/a/tKZdzrg) without opening the door, using your pet (if you got one).

Yep, I already fixed it in #7902
I don't know why ChromieCraft is taking so long to get such an important fix to the server, that's kind a loot exploit right there.

**Comment by patou01** [comment]
It seems this issue is due to the state being wrongly saved. I haven't figured out why yet.
``` 
 instance->SetData(DATA_ARCHAEDAS, IN_PROGRESS);
  instance->GetData(DATA_ARCHAEDAS);
```
doesn't seem to show the same result. ie if I set (line 227 or something like that in uldaman.cpp), it nevers makes it to the actual instance.
